packages/video_player/video_player_web/test/video_player_web_test.dart

```
@@ -82,6 +83,29 @@ void main() {
       expect(VideoPlayerPlatform.instance.play(textureId), completes);
     });
 
+    test('throws PlatformException when playing bad media', () async {
+      int videoPlayerId = await VideoPlayerPlatform.instance.create(
+        DataSource(
+            sourceType: DataSourceType.network,
+            uri:
+                'https://flutter.github.io/assets-for-api-docs/assets/videos/_non_existent_video.mp4'),
+      );
+
+      // Mute video to allow autoplay (See https://goo.gl/xX8pDD)
+      await VideoPlayerPlatform.instance.setVolume(videoPlayerId, 0);
+      Stream<VideoEvent> events =
+          VideoPlayerPlatform.instance.videoEventsFor(videoPlayerId);
+
+      try {
+        await VideoPlayerPlatform.instance.play(videoPlayerId);
+      } catch (e) {
+        // The version of Chrome we're using in test may throw a DomException here!
```

So this could be a `DomException` instead of a `PlatformException`?

---

What I've seen in the test is that it is *both*, there's an onError event being triggered, and then we funnel the PlatformException to the Stream, but also `.play` is throwing (in the browser we use for unit tests, but NOT when I test manually).

Is this maybe the 404 error being propagated up? Should we swallow this exception instead? Or try to funnel it as well to the Stream?

Probably the latter...

---

I ended up catching the promise rejection from play() and funneling it to the event Stream as a PlatformException. Documented that bit of the code as well with docs from MDN.